packages/next/src/server/api-utils/index.ts

```diff
@@ -91,8 +92,13 @@ export function checkIsOnDemandRevalidate(
   const revalidateOnlyGenerated = headers.has(
     PRERENDER_REVALIDATE_ONLY_GENERATED_HEADER
   )
+  const isISRRevalidationRequest = headers.has('x-next-isr')
```

Comment:
```suggestion
  const isISRRevalidationRequest = headers.has('x-next-stale-isr')
```
Maybe we call it `stale` ISR since that's the behavior we're going for but via the On-Demand revalidate flow

Reply:
Good point.

One thing about that is that in the implementation i've done, this stale revalidation request could be done even if `experimental.disableBackgroundRevalidation` is set to false.

This could be blocked or we could let it like that as this gives another way of doing an invalidation for app router that can use the fetch cache and is done right away. 
If we do that we probably need another name though